### ISSUE TITLE
Increase test coverage of `fuels-ts/abi-coder/coders`

### ISSUE THREAD

**Issue by danielbate** [FuelLabs/fuels-ts#761]
Per #753 to increase overall test coverage and quality.

- Missing test coverage of `B512Coder`
- Various missing throw expectations not covered in coders (eg. `ArrayCoder`, `B256Coder`, `B512Coder` etc.)

Does anyone have any opinion on potentially refactoring `packages/abi-coder/src/coders/coders.test.ts` into individual files (eg `b256.test.ts`, `b512.test.ts` etc.)? I'm a big fan of the `forEach` loops for duplication reduction, however in this instance I think verbosity could be preferred here to clearly explain the expectations for each coder. Potentially this is why more specific expectations are missed?

For example:

https://github.com/FuelLabs/fuels-ts/blob/2d70903c980a738bee82bdd1181ed84545b6aa66/packages/abi-coder/src/coders/b256.ts#L30-L32

**Comment by arboleya** [comment]
Yes, having individual tests is ideal for increased clarity and easier maintenance.

Unrelated to this, but in the future, we might also refactor the `abi-coder` package to look more like the `abi-typegen`.
